include/zephyr/drivers/stepper.h

```diff
@@ -326,6 +406,33 @@ static inline int z_impl_stepper_get_micro_step_res(const struct device *dev,
 	return api->get_micro_step_res(dev, resolution);
 }
 
+/**
+ * @brief Set the callback function to be called when a stepper fault occurs
+ *
+ * @param dev pointer to the stepper driver instance
+ * @param callback Callback function to be called when a stepper fault occurs
+ * passing NULL will disable the callback
+ * @param user_data User data to be passed to the callback function
+ *
+ * @retval -ENOSYS If not implemented by device driver
+ * @retval 0 Success
+ */
+__syscall int stepper_set_fault_callback(const struct device *dev,
```

Comment:
how is this different from `set_event_callback` ? also i don't see this API being implemented or used anywhere.